lib/internal/Magento/Framework/App/Utility/IPAddressNormaliser.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\App\Utility;
+
+use IPTools\Network;
+use IPTools\Range;
+
+/**
+ * Utility class to normalise list of IP addresses
+ */
+class IPAddressNormaliser
+{
+    /**
+     * Normalise / canonicalise list of IP addresses / ranges
+     *
+     * @param  string[] $addresses
+     * @return string[]
+     */
+    public function execute(array &$addresses): array
```

Comment:
I believe we shouldn't use reference here, it's not really logical to modify original array when you're still returning result array

Reply:
Good catch. This is left over from an architecture change I made. I'll fix this up now.